packages/ra-tree-ui-materialui/src/TreeForm.js

```diff
@@ -0,0 +1,198 @@
+    parentSource,
+    pristine,
+    pure,
+    redirect,
+    reset,
+    resetSection,
+    save,
+    startUndoable,
+    submit,
+    submitFailed,
+    submitSucceeded,
+    submitting,
+    touch,
+    translate,
+    triggerSubmit,
+    undoable,
+    undoableDragDrop,
+    untouch,
+    valid,
+    validate,
+    ...props
+}) => props;
+class Form extends Component {
+    static defaultProps = {
+        actions: <TreeFormActions />,
+    };
+
+    handleClick = event => {
+        event.persist();
+        // This ensure clicking on an input or button does not collapse/expand a node
+        // When clicking on the form (empty spaces around inputs) however, it should
+        // propagate to the parent
+        if (event.target.tagName.toLowerCase() !== 'form') {
+            event.stopPropagation();
+        }
+    };
+
+    handleDrop = event => {
+        event.persist();
+        if (this.props.cancelDropOnChildren) {
+            event.preventDefault();
+        }
+    };
+
+    handleSubmit = () => {
+        const {
+            basePath,
+            dispatchCrudUpdate,
+            handleSubmit,
+            node: { record },
+            resource,
+            startUndoable,
+            undoable = true,
+        } = this.props;
+
+        return handleSubmit(
+            values =>
+                undoable
+                    ? startUndoable(
+                          crudUpdateAction(
+                              resource,
+                              record.id,
+                              { ...record, ...values },
+                              record,
+                              basePath,
+                              false
+                          )
+                      )
+                    : dispatchCrudUpdate(
+                          resource,
+                          record.id,
+                          { ...record, ...values },
+                          record,
+                          basePath,
+                          false
+                      )
+        );
+    };
+
+    render() {
+        const {
+            actions,
+            basePath,
+            children,
+            classes,
+            handleSubmit,
+            invalid,
+            node,
+            pristine,
+            resource,
+            saving,
+            submitOnEnter = true,
+            ...props
+        } = this.props;
+
+        return (
+            <form className={classes.root} onClick={this.handleClick}>
+                {Children.map(
+                    children,
+                    field =>
+                        field
+                            ? cloneElement(field, {
+                                  basePath: field.props.basePath || basePath,
+                                  onDrop: this.handleDrop,
+                                  record: node.record,
+                                  resource,
+                                  ...sanitizeRestProps(props),
```

Comment:
I don't think you should pass the additional props to each child, but to the root element

Reply:
Some of them must be passed, I'll make it more obvious

Reply:
Actually none of the rest props were needed. I removed this line